### ISSUE TITLE
2.2.6 - Product removed from flat after save / Status issue

### ISSUE THREAD

**Issue by jburel-ca** [magento/magento2#19363]
### Preconditions (*)
1. Magento 2.2.6 version, with catalog product flat enabled
2. Add a product in a category
3. Staging must already be used: row_id and entity_id columns must have different values!

### Steps to reproduce (*)
1. Frontend: go to the category. Your product is visible!
2. Backend: go the product, and save it without changing anything

### Expected result (*)
1. Frontend: refresh your category. Your product is visible again.

### Actual result (*)
1. Frontend: refresh your category. Your product is not here anymore...
2. If you reindex catalog_product_flat index by using bin/magento binary, this product become visible in the category...

My debug:
When the product was saved, the product's `entity_id` is sent to the controller and others.
But, issue is in this file: **src/vendor/magento/module-catalog/Model/Indexer/Product/Flat/Action/Row.php**

In 2.2.6, these new lines was added (after line 94):
```php
/* @var $status \Magento\Eav\Model\Entity\Attribute */
$status = $this->_productIndexerHelper->getAttribute(ProductInterface::STATUS);
$statusTable = $status->getBackend()->getTable();
$statusConditions = [
    'store_id IN(0,' . (int)$store->getId() . ')',
    'attribute_id = ' . (int)$status->getId(),
    $linkField . ' = ' . (int)$id,
];
$select = $this->_connection->select();
$select->from($statusTable, ['value'])
    ->where(implode(' AND ', $statusConditions))
    ->order('store_id DESC')
    ->limit(1);
$result = $this->_connection->query($select);
$status = $result->fetchColumn(0);

if ($status == \Magento\Catalog\Model\Product\Attribute\Source\Status::STATUS_ENABLED) {
(....)
```

In fact, `$linkField` is equal to "row_id" because status table (`catalog_product_entity_int`) have a single link field named row_id.
But, the `$id` variable is the product `entity_id` column...
So, if the status is not found (because it's not the good product), the product's line is not inserted into flat table, and the product disappears from category, search, and other!

Thanks!

**Comment by magento-engcom-team** [comment]
Hi @jburel-clever-age. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@jburel-clever-age do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-andrii. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @jburel-clever-age , thank you for your report. Please follow these guidelines for proper tracking of your issue. You can report Commerce-related issues in one of two ways:
You can use the Support portal associated with your account
or
If you are a Partner reporting on behalf of a merchant, use the Partner portal.
 
GitHub is intended for Magento Open Source users to report on issues related to Open Source only. There are no account management services associated with GitHub.

**Comment by jburel-clever-age** [comment]
Hi,

This issue is related to a community edition, so please keep this ticket opened.....
Maybe you won't fix this, it's your business, but in fact this ticket must be opened.

Our project is not an Commerce edition in this case.

Thanks in advance..

Note: I can't edit the previous posts, so I can't "tick" the checkbox in the checklist :(

**Comment by ghost** [comment]
Hi @jburel-clever-age. This is not available for Magento Open Source: 
  3. Staging must already be used: row_id and entity_id columns must have different values!
